### PR TITLE
[DEVOPS-1333] Add code signing to the Windows bws CLI

### PR DESCRIPTION
**Pull request by michalchecinski**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [x] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Digitally sign Windows CLI .exe executable to prevent warning showing up on clients' computers while running `bws` commands.

## Code changes

- **.github/workflows/build-cli.yml:** 
  - split Windows and UNIX build jobs
  - add steps to windows build job to login to Azure, get secrets from KeyVault, install azuresigntool and use azuresigntool to sign windows CLI artifact.

## Before you submit

- Please add **unit tests** where it makes sense to do so

### CODE REVIEW

**Review by Hinton — CHANGES_REQUESTED** [review]
Please ensure the secrets in the pipeline are optional by only requiring them if we're publishing or have the secrets set, this will ensure external PRs don't automatically fail.

I wonder if we could put the signing as a separate stage after the build which would avoid repeating the build command.

**Comment by michalchecinski** [comment]
@Hinton The build command is not repeated. I split the Unix and Windows build - the steps to sign macOS and Windows are different, so to make it clear, I made different build jobs for each system. I'll make signing steps run only on main/rc branches to ensure they will work for the external PRs.

File: .github/workflows/build-cli.yml [hunk 1]
```diff
@@ -30,6 +31,103 @@ jobs:
           VERSION=$(grep -o '^version = ".*"' crates/bws/Cargo.toml | grep -Eo "[0-9]+\.[0-9]+\.[0-9]+")
           echo "package_version=$VERSION" >> $GITHUB_OUTPUT
 
+      - name: Sign if branch is main or rc
+        id: sign
+        run: |
+          if [[ $GITHUB_REF == refs/heads/main || $GITHUB_REF == refs/heads/rc ]]; then
```
Comment (vgrassia) [inline .github/workflows/build-cli.yml:37]: Should we also be checking for `hotfix-rc`?

File: .github/workflows/build-cli.yml [hunk 2]
```diff
@@ -30,6 +31,103 @@ jobs:
...
+            echo "sign=true" >> $GITHUB_OUTPUT
+          fi
+          echo "sign=false" >> $GITHUB_OUTPUT
+
+  build-windows:
+    name: Building CLI for - ${{ matrix.settings.os }} - ${{ matrix.settings.target }}
+    runs-on: ${{ matrix.settings.os || 'ubuntu-latest' }}
+    needs:
+      - setup
```
Comment (vgrassia) [inline .github/workflows/build-cli.yml:46]:
```suggestion
    needs: setup
```

**Review by vgrassia — CHANGES_REQUESTED** [review]

**Comment by Hinton** [comment]
Thanks for excluding the secrets. I'm still slightly concern with the duplication of steps, since it's easy to miss modifying one of the places.

File: .github/workflows/build-cli.yml (hunk 1 above)
Comment (Hinton) [reply .github/workflows/build-cli.yml:37]: We might also want to check if the origin is actually this repo and not a fork.

```if: github.repository_owner == 'Upvision'```

Comment (michalchecinski) [reply .github/workflows/build-cli.yml]: @Hinton That's great idea!

File: .github/workflows/build-cli.yml [hunk 3]
```diff
@@ -30,6 +31,104 @@ jobs:
           VERSION=$(grep -o '^version = ".*"' crates/bws/Cargo.toml | grep -Eo "[0-9]+\.[0-9]+\.[0-9]+")
           echo "package_version=$VERSION" >> $GITHUB_OUTPUT
 
+      - name: Sign if branch is main or rc
```
Comment (vgrassia) [inline .github/workflows/build-cli.yml:34]:
```suggestion
      - name: Sign if repo is owned by Bitwarden
```

**Review by vgrassia — CHANGES_REQUESTED** [review]

**Review by mimartin12 — APPROVED** [review]

**Review by mimartin12 — DISMISSED** [review]

**Review by mimartin12 — APPROVED** [review]

**Review by vgrassia — APPROVED** [review]

**Pull request merged**